pkg/roachpb/metadata.go

```diff
@@ -773,3 +774,28 @@ func (l Locality) AddTier(tier Tier) Locality {
 	}
 	return Locality{Tiers: []Tier{tier}}
 }
+
+// IsEmpty returns true if hint contains no data.
+func (h *GCRangeHint) IsEmpty() bool {
+	return h.EstimatedRangeRemovalTime.IsEmpty()
+}
+
+// Merge combines GC hint information about merging replicas and updates
+// LHS receiver part.
+// returns true if receiver state was changed.
+func (h *GCRangeHint) Merge(rhs *GCRangeHint) bool {
+	leftEmpty := h.EstimatedRangeRemovalTime.IsEmpty()
+	rightEmpty := rhs.EstimatedRangeRemovalTime.IsEmpty()
+	if leftEmpty || rightEmpty {
+		// If one of sides doesn't have hint we probably can't do anything sensible.
```

Comment:
Would it be better to just use the non-empty one?

Reply:
My initial thinking was that if we merge with a range that had no hint it means it will likely have data in it so it would be a waste. But maybe you are right and we can still run the range through expedited checks and probe if they are still eligible if we merged "hinted" ranges with empty ones on the range boundaries.